dali/pipeline/operator/sequence_shape.h

```diff
@@ -197,51 +198,61 @@ class UnfoldedSliceRange {
   size_t slice_stride_;
 };
 
+template <typename Type>
+TensorLayout unfolded_sample_layout(const Type &batch, int ndims_to_unfold) {
+  assert(batch.sample_dim() >= ndims_to_unfold);
+  auto sample_dim = batch.sample_dim() - ndims_to_unfold;
+  const auto &initial_layout = batch.GetLayout();
+  return initial_layout.empty() ? "" : initial_layout.last(sample_dim);
+}
+
+template <typename Type>
+void setup_expanded_like(const Type &batch, Type &expanded_batch) {
```

Comment:
This isn't a generic function that can work with anything - it has very specific requirements for the target type and that should be somehow reflected in the template parameter name.
```suggestion
template <typename BufferObject>
void setup_expanded_like(const BufferObject &batch, BufferObject &expanded_batch) {
```

Reply:
done